From 005b4da3ae1feab914a401e9bc7027397826d041 Mon Sep 17 00:00:00 2001
From: Cezary Rojewski <cezary.rojewski@intel.com>
Date: Tue, 14 Aug 2018 12:02:26 +0200
Subject: [PATCH 0714/1132] ASoC: Intel: Move kbl machine drv tables to common
 directory.

Align with machine-drv-tables declaration model present on upstream.
Tables are moved from skl.c into separate file within common directory
dedicated to kbl-specific data only.

Change-Id: I25846b83d6c791c7c7dae25766a06beddd686277
Depends-On: I6780bcd1506709757e673130ffc0ce6d0b26709f
Signed-off-by: Cezary Rojewski <cezary.rojewski@intel.com>
Reviewed-on:
Reviewed-by: Uzieblo, Olgierd <olgierd.uzieblo@intel.com>
---
 sound/soc/intel/common/soc-acpi-intel-kbl-match.c |  4 ++
 sound/soc/intel/skylake/skl.c                     | 79 -----------------------
 2 files changed, 4 insertions(+), 79 deletions(-)

diff --git a/sound/soc/intel/common/soc-acpi-intel-kbl-match.c b/sound/soc/intel/common/soc-acpi-intel-kbl-match.c
index 0ee173c..750a7f8 100644
--- a/sound/soc/intel/common/soc-acpi-intel-kbl-match.c
+++ b/sound/soc/intel/common/soc-acpi-intel-kbl-match.c
@@ -35,7 +35,11 @@ static struct snd_soc_acpi_codecs kbl_7219_98357_codecs = {
 struct snd_soc_acpi_mach snd_soc_acpi_intel_kbl_machines[] = {
 	{
 		.id = "INT343A",
+#if IS_ENABLED(CONFIG_SND_SOC_INTEL_KBLR_RT298_MACH)
+		.drv_name = "kblr_alc298s_i2s",
+#else
 		.drv_name = "kbl_alc286s_i2s",
+#endif
 		.fw_filename = "intel/dsp_fw_kbl.bin",
 	},
 	{
diff --git a/sound/soc/intel/skylake/skl.c b/sound/soc/intel/skylake/skl.c
index 875a487..e9f172d 100644
--- a/sound/soc/intel/skylake/skl.c
+++ b/sound/soc/intel/skylake/skl.c
@@ -1159,26 +1159,6 @@ static struct snd_soc_acpi_codecs skl_codecs = {
 	.codecs = {"10508825"}
 };
 
-static struct snd_soc_acpi_codecs kbl_codecs = {
-	.num_codecs = 1,
-	.codecs = {"10508825"}
-};
-
-static struct snd_soc_acpi_codecs kbl_poppy_codecs = {
-	.num_codecs = 1,
-	.codecs = {"10EC5663"}
-};
-
-static struct snd_soc_acpi_codecs kbl_5663_5514_codecs = {
-	.num_codecs = 2,
-	.codecs = {"10EC5663", "10EC5514"}
-};
-
-static struct snd_soc_acpi_codecs kbl_7219_98357_codecs = {
-	.num_codecs = 1,
-	.codecs = {"MX98357A"}
-};
-
 static struct skl_machine_pdata cnl_pdata = {
 	.use_tplg_pcm = true,
 };
@@ -1208,65 +1188,6 @@ static struct snd_soc_acpi_mach sst_skl_devdata[] = {
 	{}
 };
 
-static struct snd_soc_acpi_mach sst_kbl_devdata[] = {
-	{
-		.id = "INT343A",
-#if IS_ENABLED(CONFIG_SND_SOC_INTEL_KBLR_RT298_MACH)
-		.drv_name = "kblr_alc298s_i2s",
-#else
-		.drv_name = "kbl_alc286s_i2s",
-#endif
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-	},
-	{
-		.id = "INT343B",
-		.drv_name = "kbl_n88l25_s4567",
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-		.machine_quirk = snd_soc_acpi_codec_list,
-		.quirk_data = &kbl_codecs,
-		.pdata = &skl_dmic_data
-	},
-	{
-		.id = "MX98357A",
-		.drv_name = "kbl_n88l25_m98357a",
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-		.machine_quirk = snd_soc_acpi_codec_list,
-		.quirk_data = &kbl_codecs,
-		.pdata = &skl_dmic_data
-	},
-	{
-		.id = "MX98927",
-		.drv_name = "kbl_r5514_5663_max",
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-		.machine_quirk = snd_soc_acpi_codec_list,
-		.quirk_data = &kbl_5663_5514_codecs,
-		.pdata = &skl_dmic_data
-	},
-	{
-		.id = "MX98927",
-		.drv_name = "kbl_rt5663_m98927",
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-		.machine_quirk = snd_soc_acpi_codec_list,
-		.quirk_data = &kbl_poppy_codecs,
-		.pdata = &skl_dmic_data
-	},
-	{
-		.id = "10EC5663",
-		.drv_name = "kbl_rt5663",
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-	},
-	{
-		.id = "DLGS7219",
-		.drv_name = "kbl_da7219_max98357a",
-		.fw_filename = "intel/dsp_fw_kbl.bin",
-		.machine_quirk = snd_soc_acpi_codec_list,
-		.quirk_data = &kbl_7219_98357_codecs,
-		.pdata = &skl_dmic_data
-	},
-
-	{}
-};
-
 static struct snd_soc_acpi_mach sst_glk_devdata[] = {
 	{
 		.id = "INT343A",
-- 
2.7.4

